interfaces/apparmor/spec.go

```diff
@@ -42,6 +43,51 @@ const (
 	UnconfinedEnabled
 )
 
+type prioritizedSnippets struct {
+	priority uint
+	list     []string
+}
+
+// SnippetKey is an opaque string identifying a class of snippets.
+//
+// Some APIs require the use of snippet keys to allow adding many different snippets
+// with the same key but possibly different priority.
+type SnippetKey struct {
+	key string
+}
+
+func (pk *SnippetKey) String() string {
+	return pk.key
+}
+
+func newSnippetKey(key string) SnippetKey {
+	return SnippetKey{key: key}
+}
+
+// registeredKeys is a list of allowed keys for prioritized snippets.
+// Trying to add a prioritized snippet with an unregistered key is
+// an error. Trying to register the same key twice is also an error.
+var registeredKeys map[SnippetKey]bool = make(map[SnippetKey]bool)
+
+// RegisterSnippetKey adds a key to the list of valid keys
+func RegisterSnippetKey(key string) SnippetKey {
+	snippetKey := newSnippetKey(key)
+	if _, ok := registeredKeys[snippetKey]; ok {
+		logger.Panicf("priority key %s is already registered", key)
+	}
+	registeredKeys[snippetKey] = true
+	return snippetKey
+}
+
+// GetSnippetKey retrieves all the current valid keys
+func GetSnippetKeys() []string {
+	keylist := make([]string, 0)
```

Comment:
There's no need to make an empty slice, but making one with the right capacity is useful to avoid re-allocations:

```suggestion
	keylist := make([]string, 0, len(registeredKeys))
```

Reply:
Good point. Done.